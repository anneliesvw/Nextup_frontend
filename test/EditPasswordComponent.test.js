import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';
import EditPasswordComp from '../src/components/users/dialogs/EditPassword.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('EditPasswordComp', () => {
  let store;
  let actions;
  beforeEach(() => {
    actions = {
      updatePassword: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  // faalt?

  it('dispatches an action when a button is clicked', () => {
    const wrapper = shallow(EditPasswordComp, {
      store,
      localVue,
      data: {
        userInfo: {
          person: { firstName: 'Jef', lastName: 'Test', location: { name: 'TestLocation' } },
          username: 'TestUser',
        },
      },
    });
    wrapper.find('.update-password-btn').trigger('click');
    expect(actions.updatePassword.mock.calls).toHaveLength(1);
  });
});
