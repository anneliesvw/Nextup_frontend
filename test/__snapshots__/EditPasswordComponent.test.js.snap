// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPasswordComp should match snapshot 1`] = `
<el-dialog
  class="edit-password-dialog"
  title="Edit password"
>
  <div
    class="edit-password-form"
  >
    <el-form
      class="edit-password"
      label-position="top"
    >
      <el-form-item
        label="old password"
      >
        <div
          class="edit-password-group"
        >
          <el-input
            class="edit-password-input"
            type="password"
          />
           
          <el-button
            class="edit-password-btn"
            type="info"
          >
            <div
              class="edit-password-img"
            >
              <img
                alt="monkey"
                class="edit-password-monkey1"
                src="https://cdn.thinglink.me/api/image/1010252198820446210/1240/10/scaletowidth"
              />
            </div>
          </el-button>
        </div>
      </el-form-item>
       
      <el-form-item
        label="new password"
      >
        <div
          class="edit-password-group"
        >
          <el-input
            class="edit-password-input"
            type="password"
          />
           
          <el-button
            class="edit-password-btn"
            type="info"
          >
            <div
              class="edit-password-img"
            >
              <img
                alt="monkey"
                class="edit-password-monkey1"
                src="https://cdn.thinglink.me/api/image/1010252198820446210/1240/10/scaletowidth"
              />
            </div>
          </el-button>
        </div>
      </el-form-item>
       
      <el-form-item
        label="new password again"
      >
        <div
          class="edit-password-group"
        >
          <el-input
            class="edit-password-input"
            type="password"
          />
           
          <el-button
            class="edit-password-btn"
            type="info"
          >
            <div
              class="edit-password-img"
            >
              <img
                alt="monkey"
                class="edit-password-monkey1"
                src="https://cdn.thinglink.me/api/image/1010252198820446210/1240/10/scaletowidth"
              />
            </div>
          </el-button>
        </div>
         
        <!---->
      </el-form-item>
    </el-form>
  </div>
   
  <span
    class="dialog-footer"
    slot="footer"
  >
    <el-button>
      Cancel
    </el-button>
     
    <el-button
      class="update-password-btn"
      type="primary"
    >
      Save
    </el-button>
  </span>
</el-dialog>
`;
